### ISSUE TITLE
ERROR: Dependency "pylon" not found, tried cmake 

### ISSUE THREAD

**Issue by jahwanoh** [basler/gst-plugin-pylon#15]
Hi,

Following this intruction to use it as a the local gstreamer's plugin
https://github.com/basler/gst-plugin-pylon#integrating-with-gstreamer-monorepo

it never success to configuration with this script.
PYLON_ROOT=/opt/pylon meson builddir --prefix /usr -Dcustom_subprojects=gst-plugin-pylon

Here is the log

Executing subproject gst-plugin-pylon 

gst-plugin-pylon| Project name: gst-plugin-pylon
gst-plugin-pylon| Project version: 0.3.0
gst-plugin-pylon| C compiler for the host machine: cc (gcc 7.5.0 "cc (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0")
gst-plugin-pylon| C linker for the host machine: cc ld.bfd 2.30
gst-plugin-pylon| C++ compiler for the host machine: c++ (gcc 7.5.0 "c++ (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0")
gst-plugin-pylon| C++ linker for the host machine: c++ ld.bfd 2.30
gst-plugin-pylon| Dependency gstreamer-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Message: Building against GStreamer 1.21.0.1
gst-plugin-pylon| Compiler for C supports link arguments -Wl,-Bsymbolic-functions: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -fvisibility=hidden: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -fno-strict-aliasing: YES (cached)
gst-plugin-pylon| Message: Disabling deprecated GLib API
gst-plugin-pylon| Checking for size of "char" : 1
gst-plugin-pylon| Checking for size of "int" : 4
gst-plugin-pylon| Checking for size of "long" : 8
gst-plugin-pylon| Checking for size of "short" : 2
gst-plugin-pylon| Checking for size of "void*" : 8
gst-plugin-pylon| Checking for size of "off_t" : 8
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-declarations: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wmissing-declarations: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wredundant-decls: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wredundant-decls: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wwrite-strings: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wwrite-strings: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Winit-self: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Winit-self: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-include-dirs: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wmissing-include-dirs: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wno-multichar: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wno-multichar: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wvla: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wvla: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wpointer-arith: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wpointer-arith: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-prototypes: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wdeclaration-after-statement: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wold-style-definition: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Waggregate-return: YES (cached)
gst-plugin-pylon| Dependency gstreamer-base-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-check-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-pbutils-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-allocators-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-video-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency glib-2.0 found: YES 2.72.2 (overridden)
gst-plugin-pylon| Dependency gobject-2.0 found: YES 2.72.2 (overridden)
gst-plugin-pylon| Message: GStreamer debug system is enabled
gst-plugin-pylon| Program python3 found: YES (/usr/bin/python3.7)
gst-plugin-pylon| Run-time dependency pylon found: NO (tried cmake)
gst-plugin-pylon| Message: fallback to detect pylon 6.x
gst-plugin-pylon| Run-time dependency pylon found: NO (tried cmake)

subprojects/gst-plugin-pylon/ext/pylon/meson.build:25:4: ERROR: Dependency "pylon" not found, tried cmake

**Comment by thiesmoeller** [comment]
There is an issue with using gstreamer head in monorepo configuration.

If you checkout older release tags this is working.
We'll investigate

**Comment by jahwanoh** [comment]
I will try with [v0.2.0](https://github.com/basler/gst-plugin-pylon/releases/tag/v0.2.0) release.
Thank you.

**Comment by jahwanoh** [comment]
v0.2.0 also gives error as below.
Could you let me know which branch or commit would be working fine?

Executing subproject gst-plugin-pylon 

gst-plugin-pylon| Project name: gst-plugin-pylon
gst-plugin-pylon| Project version: 0.2.0
gst-plugin-pylon| C compiler for the host machine: cc (gcc 7.5.0 "cc (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0")
gst-plugin-pylon| C linker for the host machine: cc ld.bfd 2.30
gst-plugin-pylon| C++ compiler for the host machine: c++ (gcc 7.5.0 "c++ (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0")
gst-plugin-pylon| C++ linker for the host machine: c++ ld.bfd 2.30
gst-plugin-pylon| Dependency gstreamer-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Message: Building against GStreamer 1.21.0.1
gst-plugin-pylon| Compiler for C supports link arguments -Wl,-Bsymbolic-functions: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -fvisibility=hidden: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -fno-strict-aliasing: YES (cached)
gst-plugin-pylon| Message: Disabling deprecated GLib API
gst-plugin-pylon| Checking for size of "char" : 1
gst-plugin-pylon| Checking for size of "int" : 4
gst-plugin-pylon| Checking for size of "long" : 8
gst-plugin-pylon| Checking for size of "short" : 2
gst-plugin-pylon| Checking for size of "void*" : 8
gst-plugin-pylon| Checking for size of "off_t" : 8
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-declarations: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wmissing-declarations: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wredundant-decls: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wredundant-decls: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wwrite-strings: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wwrite-strings: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Winit-self: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Winit-self: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-include-dirs: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wmissing-include-dirs: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wno-multichar: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wno-multichar: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wvla: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wvla: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wpointer-arith: YES (cached)
gst-plugin-pylon| Compiler for C++ supports arguments -Wpointer-arith: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wmissing-prototypes: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wdeclaration-after-statement: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Wold-style-definition: YES (cached)
gst-plugin-pylon| Compiler for C supports arguments -Waggregate-return: YES (cached)
gst-plugin-pylon| Dependency gstreamer-base-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-check-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-pbutils-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-allocators-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency gstreamer-video-1.0 found: YES 1.21.0.1 (overridden)
gst-plugin-pylon| Dependency glib-2.0 found: YES 2.72.2 (overridden)
gst-plugin-pylon| Dependency gobject-2.0 found: YES 2.72.2 (overridden)

subprojects/gst-plugin-pylon/meson.build:205:1: ERROR: Dependencies must be external dependencies

**Comment by jahwanoh** [comment]
after resolving `subprojects/gst-plugin-pylon/meson.build:205:1: ERROR: Dependencies must be external dependencies ` issue, it also gives `ERROR: Dependency "pylon" not found, tried cmake`

**Comment by michaelgruner** [comment]
The integration with GStreamer monorepo is working with Pylon 7. We are working on a fix for Pylon 6. I'll keep you updated.

**Comment by thiesmoeller** [comment]
this should fix:

```patch
diff --git a/ext/pylon/meson.build b/ext/pylon/meson.build
index a25cf1d..81ac3ec 100644
--- a/ext/pylon/meson.build
+++ b/ext/pylon/meson.build
@@ -22,7 +22,7 @@ if pylon_dep.found()
   endif
 else
     message('fallback to detect pylon 6.x')
-    pylon_dep = dependency('pylon', method : 'cmake', cmake_module_path: 'cmake', required: true)
+    pylon_dep = dependency('pylon', method : 'cmake', cmake_module_path: meson.project_source_root() / 'cmake', required: true)
     # runtime path is properly defined by cmake
     pylon_link_args = []
 endif
```

**Comment by michaelgruner** [comment]
A fix has been upstreamed. It will be included in the incoming release, hopefully early next week.

**Comment by thiesmoeller** [comment]
fixed in v0.4.0
